content/docs/gke/private-clusters.md

```diff
@@ -7,65 +7,208 @@ weight = 5
 This guide describes how to create private clusters with Kubeflow on Google 
 Kubernetes Engine (GKE).
 
+Creating a [private Kubernetes Engine cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters)
+means the Kubernetes Engine nodes won't have public IP addresses. This can improve security by blocking unwanted outbound/inbound
+access to nodes. Removing IP addresses means external services (including GitHub, PyPi, DockerHub etc...) won't be accessible
+from the nodes. Google services (BigQuery, Cloud Storage, etc...) are still accessible.
+
+Importantly this means you can continue to use your [Google Container Registry (GCR)](https://cloud.google.com/container-registry/docs/) to host your Docker images. Other Docker registries (for example DockerHub) will not be accessible. If you need to use Docker images
+hosted outside Google Container Registry you can use the scripts provided by Kubeflow
+to mirror them to your GCR registry.
+
 ## Before you start
 
-This guide assumes you have already set up Kubeflow with GKE. If you haven't done
-so, follow the guide to [deploying Kubeflow on GCP](/docs/gke/deploy/).
+Before installing Kubeflow ensure you have installed the following tools:
+    
+  * [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+  * [gcloud](https://cloud.google.com/sdk/)
+  * [jsonnet](https://github.com/google/jsonnet/releases)
 
-## Private clusters
+## Deploy Kubeflow with Private GKE
 
-Creating a [private Kubernetes Engine cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters)
-means the Kubernetes Engine nodes won't have public IP addresses. This can improve security by blocking unwanted outbound/inbound
-access to nodes. Removing IP addresses means external services (including GitHub, PyPi, DockerHub etc...) won't be accessible
-from the nodes. Google services (for example, Container Registry) are still accessible.
+1. Set user credentials. You only need to run this command once:
+   
+    ```
+    gcloud auth application-default login
+    ```
+1. Copy non-GCR hosted images to your GCR registry
+
+    1. Clone the Kubeflow source 
+
+        ```
+        git clone https://github.com/kubeflow/kubeflow.git git_kubeflow      
+        ```
+    1. Use [Google Cloud Builder(GCB)](https://cloud.google.com/cloud-build/docs/) to replicate the images
+
+        ```
+        cd ~/git_kubeflow/scripts/gke
+        PROJECT=<PROJECT> make copy-gcb
```

Comment:
nit - is images in cloud-ml-dev already public? if so maybe this step can be omitted?

Reply:
Images in cloud-ml-dev are not public. Users need to substitute in their own project; they shouldn't be using cloud-ml-dev.